docs/types-and-interfaces.md

```diff
@@ -0,0 +1,53 @@
+# TypeScript Types & Interfaces 🧩
+
+This document explains the **types and interfaces** used in `linid-im-front-corelib`.
+
+---
+
+## 🔌 LinidZoneEntry
+
+Represents a plugin entry that can be rendered inside a zone.
+
+```ts
+export interface LinidZoneEntry {
+  /** Path or identifier of the plugin module */
+  plugin: string;
+
+  /** Props forwarded to the plugin component */
+  props?: Record<string, unknown>;
+}
+````
```

Comment:
Code fence has four backticks instead of three. This should be three backticks (```) to properly close the TypeScript code block.